### DISCUSSION TITLE
How to configure kitty v0.25.0 to not prompt for confirmation while closing a tab?

### DISCUSSION THREAD

**Discussion by jarmos-san** [kovidgoyal/kitty#4956] [Q&A] [answered]
I've updated Kitty to it's recent version i.e v0.25.0 & it kinda "broke" my workflow (more like muscle memory). So, I was wondering if there was any way to configure Kitty to not prompt for a confirmation when I'm closing tabs?

I open multiple tabs & when I'm closing one or multiple tabs, I'm pretty sure I want to close them. There's no need for me to keep reminding me with a confirmation. Having an option to be able configure Kitty to revert back to it's earlier behaviour would be such a godsend for me now.

**Comment by page-down** [comment, marked as answer]
> ... configure Kitty to prompt for a confirmation when I'm closing tabs ...

Maybe you can try the following option.

https://sw.kovidgoyal.net/kitty/conf/#opt-kitty.confirm_os_window_close

```conf
confirm_os_window_close 1
```

EDIT:
If you do not want the closing confirmation.

```conf
confirm_os_window_close 0
```

**Reply by johnallen3d** [reply]
I ran into this as well. Setting `confirm_os_window_close 0` did the trick, thanks @page-down! It looks like this setting has existed for quite some time, I wonder if the default change with `v0.25.0`.

**Reply by page-down** [reply]
> ... Setting `confirm_os_window_close 0` did the trick ... I wonder if the default change with `v0.25.0`.

Yes, this change was missed in the changelog.

https://github.com/kovidgoyal/kitty/commit/00c1802319083c0176fc33d40960aca45f0efd98

With shell integration enabled by default, the new default value `-1` will show a prompt when not sitting at a shell prompt.

**Reply by johnallen3d** [reply]
Thanks again @page-down!

**Comment by adamnejm** [comment]
While I get that this thing could be useful to some people, I wish future features that change workflow in drastic ways would not be enabled by default.

**Reply by port19x** [reply]
Or at least communicated well in advance with the reasoning clearly laid out

**Comment by kovidgoyal** [comment]
confirm_os_window_close 0
